### PR TITLE
ci: Use next tag for latest commit builds

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Use next tag for doing nightly builds (create a duplicated job, so that we can retire the nightly tag separately when we are ready)
Replace occurances of "nightly" with "next"
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19291

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by nickboldt — APPROVED** [review]
LGTM, let's try it and see what explodes. :)

**Comment by nickboldt** [comment]
LOVE that you also removed all the obsolete refs to centosci :D

**Review by JPinkney — APPROVED** [review]

File: .github/workflows/next-build.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2020-2021 Red Hat, Inc.
```
Comment (benoitf) [inline .github/workflows/next-build.yml:2]:
```suggestion
# Copyright (c) 2021 Red Hat, Inc.
```

File: README.md [hunk 2]
```diff
@@ -1,8 +1,3 @@
-[![Master Build Status](https://ci.centos.org/buildStatus/icon?subject=master&job=devtools-che-devfile-registry-build-master/)](https://ci.centos.org/job/devtools-che-devfile-registry-build-master/)
```
Comment (benoitf) [inline README.md:1]: could you please use links to github actions instead of removing it ?

**Review by benoitf — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mkuznyetsov) [reply README.md:1]: sure

**Review by ericwill — APPROVED** [review]

**Pull request merged**